backend/src/test/java/com/onlinebookstore/service/BookControllerTest.java

```diff
@@ -130,6 +133,90 @@ void getBookById_NotFound() {
                 "Controller should respond with HttpStatus.NOT_FOUND");
     }
 
+    @Test
+    void deleteByIdSuccess() {
```

Comment:
overall is fine but please extend coverage
minimum tests amount is:
1 default positive case (already present)
1 case where operation is requested for an inexisting entity (delete id that is not present, excessive filters on search etc)
? possibly additional cases for situations where you may assume additional handling is required (missing some data in incoming dto etc, )